### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by davidpdrsn**
Thought I might as well fix the clippy warnings. Feel free to close this if you don't want the git churn.

### CODE REVIEW

**Review by hawkw — APPROVED** [review]

File: tower/src/balance/p2c/service.rs [hunk 1]
```diff
@@ -78,6 +78,7 @@ enum Error<E> {
     Canceled,
 }
 
+#[allow(clippy::len_without_is_empty)]
```
Comment (hawkw) [inline tower/src/balance/p2c/service.rs:81]: should we just consider adding an `is_empty` method?

**Review by hawkw — APPROVED** [review]

**Comment by hawkw** [comment]
One other note: if we're going to start caring about clippy lints (which I think we ought to), I think we should probably add a Clippy check step on CI. This will stop us from going and adding new code that clippy doesn't like, and having to do another big "fix clippy warnings" PR to make them go away. We can do that separately, though.

File: tower/src/balance/p2c/service.rs (hunk 1 above)
Comment (davidpdrsn) [reply tower/src/balance/p2c/service.rs]: Sure. I just wasn't sure if it was appropriate here. Not very familiar with this API. Added `is_empty` in 58821d7826a07

**Review by hawkw — APPROVED** [review]

File: tower/src/balance/p2c/service.rs [hunk 2]
```diff
@@ -107,6 +107,11 @@ where
     pub fn len(&self) -> usize {
         self.services.len()
     }
+
+    /// Returns whether or not the balancer is empty.
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
```
Comment (olix0r) [inline tower/src/balance/p2c/service.rs:113]: I would expect this to trigger clippy. Should this be:
```suggestion
        self.services.is_empty()
```

Comment (olix0r) [reply tower/src/balance/p2c/service.rs:113]: I'm not sure if ReadyCache actually has an `is_empty`, but it probably should if it does not...

Comment (davidpdrsn) [reply tower/src/balance/p2c/service.rs]: Uh good catch. Fixed in 5dc604458ccf7

**Pull request merged**